Final capstone: backend

mohamedgamaleldin999999 opened the pull request:
Book an Appointment: Complete project
In this pull request we completed the following requirements
The user logs in to the website, by typing the username and password

In the navigation panel, the user can see links to:

Motorcycles/doctors/classes/items that you selected as a theme.
"Reserve" form.
"My reservations".
"Add Doctor (the link is visible to everybody).
"Delete Doctor that you selected as a theme" (the link is visible to everybody).
On the main page, the user can see a list of Doctors.

When the user selects a specific Doctor, they can see the details page with its full description.

In the details page, the user can click the "Reserve" button.
When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

Make the app responsive, creating both mobile and desktop versions.

When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.

When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).

Use the design based on the ["Book a vespa test-ride"](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005583) and add all necessary inputs.
The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.
When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).

Add full documentation for your API.

Front-end Pull request
https://github.com/stino-x/doc-appointment/pull/19

stino-x committed:
setting up the MVC stuff

stino-x committed:
intrduction of CORS for managing ports and installation of linters

stino-x committed:
setup everything and added the groundwork for authentication

stino-x committed:
set all the code

mohamedgamaleldin999999 committed:
Fix linters

mohamedgamaleldin999999 committed:
Fix linters

mohamedgamaleldin999999 committed:
Fix linters

stino-x committed:
Merge pull request #1 from stino-x/set-up

Set up

mohamedgamaleldin999999 committed:
Update controllers

mohamedgamaleldin999999 committed:
Update appointment model

mohamedgamaleldin999999 committed:
Update teacher model

mohamedgamaleldin999999 committed:
Update user model

stino-x committed:
i made changes to our controllers suit our plans

stino-x committed:
adjusted routes for our frontend

stino-x committed:
changes to some attributes for each model

stino-x committed:
evertything wrapped up

stino-x committed:
Update to doctors, reservations, sessions, and users controllers. Also updated doctor and reservation models.

stino-x committed:
Update slot availability check and add months array to Doctor model.

mohamedgamaleldin999999 committed:
Merge pull request #2 from stino-x/logic

Logic

mohamedgamaleldin999999 committed:
Add readme

mohamedgamaleldin999999 committed:
Merge pull request #3 from stino-x/Finalize-project

Add readme

Meltrust commented on db/seeds.rb:
- [ ] This seeds file does not work:

![image](https://github.com/stino-x/doc-appointment-backend/assets/57421823/755ef6a2-0ef3-47e4-92df-98fa4ac7ac37)

Kindly, either fix its contents or remove them. 👍

Meltrust commented on README.md:
````diff
@@ -1 +1,230 @@
+  rails s
+```
+- SETUP [Front-end](https://github.com/stino-x/doc-appointment) 
+
````
- [ ] Kindly, include in your instructions the required db commands that your users will need to run your backend app: `rails db:create, rails db:migrate`. 👍

Meltrust reviewed (changes requested):
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!









#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

stino-x committed:
The commit adds the following changes:

- Adds `rails db:create` and `rails db:migrate` commands to the backend setup instructions.
- Removes the seeds file from the backend.

Meltrust commented on README.md:
```diff
@@ -1 +1,232 @@
-# teachers-appointment-backend
+# README
```
- [ ] Kindly, check the review comment on the [frontend](https://github.com/stino-x/doc-appointment/pull/19#pullrequestreview-1937872124) review.

Meltrust reviewed (changes requested):
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!









#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

Meltrust reviewed (approved):
# **Wow, you did it 🎉**
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
![applause](https://media.giphy.com/media/srg19CG0cKMuI/giphy.gif)

### To highlight


- Superb job ✔️  



Cheers and Happy coding!👏👏👏


------
